arch/arc/core/arc_smp.c

```diff
@@ -106,6 +107,8 @@ void z_arc_slave_start(int cpu_num)
 
 	arc_core_private_intc_init();
 
+	arc_irq_offload_init_smp();
```

Comment:
why not conditional on CONFIG_IRQ_OFFLOAD?

Reply:
It's done internally in `arc_irq_offload_init_smp` to avoid bringing extra irq_offload logic outside of it's files.

As `arc_irq_offload_init_smp` is defined as `static inline` this shouldn't produce extra code.

Reply:
Code size was not so much my concern, more understandability. But that's subjective (why avoiding CONFIG_IRQ_OFFLOAD dependency as a global config, but introducing arc_irq_offload_init_smp as a new possibly empty dependency? With the additional #ifdef here, to me it is more clear the offload is optional/conditional.).

Reply:
Having `arc_irq_offload_init_smp` wraper allows us to not bring internal `irq_offload` logic to the place where we just call multiple init function for secondary CPU cores.

If we want to avoid that wraper we need to add ifdefs around init call itself and we need to add ifdef'ed external function declaration.

For one subsystem it won't hurt but we may end up with what we have in kernel/init.c - there are tons of ifdefs everywhere it's not an example of well readable code, IMO.

Of course it's kinda subjective - so, if you insist I can drop wraper and move ifdefs to the calling place. But I'd prefer to keep internal logic in the subsystem code.

Reply:
I don't insist, my only point is that by calling the 'wrapper' you already bring in irq_offload logic into this function. Putting the #if here avoids the #if's in the implementation file, as including the implementation can be conditionally in the cmake file. But, this all is minor - I'm ok to remove 'draft' and approve this PR now.